site/docs/extensions/index.md

```diff
@@ -98,6 +98,34 @@ Rather than using a full data type representation, the input argument types (`sh
 | `extract(required enumeration, timestamp) => i64` | `extract:req_ts` |
 | `sum(any1) => any1`                               | `sum:any`        |
 
+### Any Types
+
+```yaml
+scalar_functions:
+- name: foo
+  impls:
+  - args:
+    - name: a
+      value: any
+    - name: b
+      value: any
+    return: int64
+```
+
+The `any` type indicates that the argument can take any possible type. In the `foo` function above, arguments `a` and `b` can be of any type, even different ones in the same function invocation.
+
+```yaml
+scalar_functions:
+- name: bar
+  impls:
+  - args:
+    - name: a
+      value: any1
+    - name: b
+      value: any1
+    return: int64
+```
+The `any[\d]` types (i.e. `any1`, `any2`, etc) impose an additional restriction. Within a single function invocation, all any types with same numeric suffix _must_ be of the same type. In the `bar` function above, arguments `a` and `b` can be bound to any values within a specific function invocation as long as both values have the same type.
```

Comment:
The "bound to any values within a specific function invocation" wording seems a bit confusing.  Maybe...

"arguments `a` and `b` can have any type as long as both types are the same"?

Reply:
Done